### PR TITLE
feat: add stub and nf-test for nanoplot module

### PR DESCRIPTION
**Pull request by awgymer**
## PR checklist

Adds a `stub` for `nanoplot` module and an nf-test for it. 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/nanoplot/tests/main.nf.test [hunk 1]
```diff
@@ -63,7 +63,41 @@ nextflow_process {
...
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(
+                            process.out.txt,
+                            process.out.html,
+                            process.out.log,
+                            process.out.versions
+                         ).match()
```
Comment (mashehu) [inline modules/nf-core/nanoplot/tests/main.nf.test:94]:
```suggestion
                { assert snapshot(process.out).match()
```
why not like this?

Comment (awgymer) [reply modules/nf-core/nanoplot/tests/main.nf.test:94]: Does it matter that `process.out.png` is empty (well optional and empty)?

Comment (mashehu) [reply modules/nf-core/nanoplot/tests/main.nf.test:94]: not for stubs

**Comment by awgymer** [comment]
Ok, it wasn't picking up the latest commit so I tried to force it by switching the target branch. Note: do not do this with the automated review requests now 🤦

**Comment by mashehu** [comment]
will-be-alright-gif

**Comment by awgymer** [comment]
I think I broke the checks too 😬

**Comment by mashehu** [comment]
Nope, github is having a moment in general...

File: modules/nf-core/nanoplot/tests/main.nf.test [hunk 2]
```diff
@@ -63,7 +63,36 @@ nextflow_process {
...
+                }
+            )
+        }
+
+    }
+
+    test("NanoPlot - stub") {
+
+        options "-stub"
+        when {
```
Comment (jasmezz) [inline modules/nf-core/nanoplot/tests/main.nf.test:75]:
```suggestion

        when {
```

File: modules/nf-core/nanoplot/tests/main.nf.test [hunk 3]
```diff
@@ -63,7 +63,36 @@ nextflow_process {
...
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match()
+                },
+                {
+                    with(process.out.html.get(0)) {
+                        assert get(1).collect { p -> file(p).getName() }.contains("NanoPlot-report.html")
+                    }
+                }
```
Comment (jasmezz) [inline modules/nf-core/nanoplot/tests/main.nf.test:95]:
```suggestion
                { assert snapshot(process.out).match() }
```

Checking for the html file seems redundant.

**Review by jasmezz — APPROVED** [review]
If you accept my small change in the `main.nf.test`, please update the snap file accordingly. Otherwise LGTM :+1:

File: modules/nf-core/nanoplot/tests/main.nf.test (hunk 3 above)
Comment (awgymer) [reply modules/nf-core/nanoplot/tests/main.nf.test]: Seems that the snapshot does not need to change for this

**Pull request merged**